server/lib/api/address.js

```diff
@@ -1,38 +1,121 @@
-module.exports = function addressrouter(router) {
+const Block = require('../../models/block.js');
+const logger = require('../logger');
+
+const MAX_BLOCKS = 200;
+
+// Shoe horned in. Not dry, also in blocks. Make db api later
+function getBlock(params, options, limit, cb) {
+  const defaultOptions = { _id: 0 };
+
+  if (!Number.isInteger(limit)) {
+    limit = MAX_BLOCKS;
+  }
+
+  Object.assign(defaultOptions, options);
+
+  Block.find(
+    params,
+    defaultOptions,
+    cb)
+    .sort({ height: -1 })
+    .limit(limit);
+}
+
+module.exports = function AddressAPI(router) {
   router.get('/addr/:addr', (req, res) => {
-    res.send(req.params.addr);
+    res.json({
+      addrStr:                 req.params.addr,
+      balance:                 0,
```

Comment:
I can definitely drop them but I am concerned about opening up the ui to property of undefined errors in case my enum of its needs is incorrect. How can I address that or is it an acceptable risk for now?